src/evict/evict_lru.c

```diff
@@ -2499,18 +2518,35 @@ __wt_page_evict_urgent(WT_SESSION_IMPL *session, WT_REF *ref)
     WT_ASSERT(session, !__wt_ref_is_root(ref));
 
     page = ref->page;
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) || S2BT(session)->evict_disabled > 0)
+    if (S2BT(session)->evict_disabled > 0 || F_ISSET_ATOMIC_16(page, WT_PAGE_IN_URGENT_QUEUE))
         return (false);
 
-    /* Append to the urgent queue if we can. */
     cache = S2C(session)->cache;
+    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) && F_ISSET(cache, WT_CACHE_EVICT_ALL))
+        return (false);
+
+    /* Append to the urgent queue if we can. */
     urgent_queue = &cache->evict_queues[WT_EVICT_URGENT_QUEUE];
     queued = false;
 
     __wt_spin_lock(session, &cache->evict_queue_lock);
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU) || S2BT(session)->evict_disabled > 0)
+
+    /* Check again, in case we raced with another thread. */
+    if (S2BT(session)->evict_disabled > 0 || F_ISSET_ATOMIC_16(page, WT_PAGE_IN_URGENT_QUEUE))
         goto done;
 
+    /*
+     * If the page is already in the LRU eviction list, clear it from the list if eviction server is
+     * not running.
+     */
+    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU)) {
+        if (!F_ISSET(cache, WT_CACHE_EVICT_ALL)) {
+            __evict_list_clear_page_locked(session, ref, true);
```

Comment:
Here, we are giving `true` as the last arg for `__evict_list_clear_page_locked`? Why are we trying to remove it from the urgent queue? We checked for the flag `WT_PAGE_IN_URGENT_QUEUE` just before, we should be sure it's not in there.

Reply:
passing true means excluding the urgent queue.

Reply:
Sorry my bad!